server/app/services/applicant/ApplicantService.java

```diff
@@ -228,14 +230,19 @@ public CompletionStage<ReadOnlyApplicantProgramService> getReadOnlyApplicantProg
   public ReadOnlyApplicantProgramService getReadOnlyApplicantProgramService(
       ApplicationModel application, ProgramDefinition programDefinition) {
     return new ReadOnlyApplicantProgramServiceImpl(
-        jsonPathPredicateGeneratorFactory, application.getApplicantData(), programDefinition);
+        jsonPathPredicateGeneratorFactory,
+        application.getApplicant(),
+        application.getApplicantData(),
+        programDefinition);
   }
 
   /** Get a {@link ReadOnlyApplicantProgramService} from applicant data and a program definition. */
   public ReadOnlyApplicantProgramService getReadOnlyApplicantProgramService(
       ApplicantData applicantData, ProgramDefinition programDefinition) {
+    ApplicantModel applicant = new ApplicantModel();
```

Comment:
not blocking since we've talked at length about this but just reiterating that I don't love that these overloads of `getReadOnlyApplicantProgramService()` actually have different behavior